src/Sentry.Unity.Editor/Android/DebugSymbolUpload.cs

```diff
@@ -1,59 +1,109 @@
 using System.IO;
+using Sentry.Extensibility;
 using Sentry.Unity.Integrations;
-using UnityEngine;
+using UnityEditor;
 
 namespace Sentry.Unity.Editor.Android
 {
-    internal static class DebugSymbolUpload
+    internal class DebugSymbolUpload
     {
-        internal static readonly string GradleExportedSymbolsPath = Path.Combine("unityLibrary", "src", "main", "symbols");
+        private readonly IDiagnosticLogger _logger;
 
-        public static string GetSymbolsPath(string unityProjectPath, string gradleProjectPath, bool isProjectExporting)
+        private readonly bool _isNewBuildingBacked;
+
+        private const string RelativeBuildOutputPathOld = "Temp/StagingArea/symbols";
+        private const string RelativeGradlePathOld = "Temp/gradleOut";
+        private const string RelativeBuildOutputPathNew = "Library/Bee/artifacts/Android";
+        private const string RelativeAndroidPathNew = "Library/Android";
+
+        private readonly string _unityProjectPath;
+        private readonly string _gradleProjectPath;
+
+        private string[] _symbolUploadPaths;
+
+        public DebugSymbolUpload(IDiagnosticLogger logger,
+            string unityProjectPath,
+            string gradleProjectPath,
+            bool isExporting,
+            IApplication? application = null)
         {
-            var symbolsPath = Path.Combine(unityProjectPath, "Temp", "StagingArea", "symbols");
-            if (isProjectExporting)
-            {
-                // We can no longer trust the Unity temp directory to exist so we copy the symbols to the export directory.
-                var copiedSymbolsPath = Path.Combine(gradleProjectPath, GradleExportedSymbolsPath);
-                CopySymbols(symbolsPath, copiedSymbolsPath);
+            application ??= ApplicationAdapter.Instance;
+
+            _logger = logger;
+
+            _unityProjectPath = unityProjectPath;
+            _gradleProjectPath = gradleProjectPath;
 
-                return copiedSymbolsPath;
+            // Starting from 2021.2 Unity caches the build output inside 'Library' instead of 'Temp'
+            if (application.UnityVersion.StartsWith("2022")
+                || application.UnityVersion.StartsWith("2021.3")
```

Comment:
This will break when 2023 comes out?
Best to try to have this inverted like: if it's not 2018, 2019, 2020 or 2021.1, then it's the new stuff.

.NET Has a class that handled Version (int he format \d\.\d\.\d\.\d\ though but might work here. https://docs.microsoft.com/en-us/dotnet/api/system.version?view=net-6.0

So you can do something like `var version = new Version(application.UnityVersion)` and then compare:
`version >= new Version("2021.1")`

Otherwise you'd need to split by `.` and check the individual parts, perhaps easier.

Reply:
TIL. Thanks!